crates/rethnet_eth/src/remote/client.rs

```diff
@@ -226,67 +300,148 @@ impl RpcClient {
             .map_err(RpcClientError::CorruptedResponse)
     }
 
-    fn serialize_request(&self, input: &MethodInvocation) -> SerializedRequest {
-        let id = jsonrpc::Id::Num(self.next_id.fetch_add(1, Ordering::Relaxed));
-        let request = serde_json::json!(Request {
-            version: crate::remote::jsonrpc::Version::V2_0,
-            id: id.clone(),
-            method: input.clone(),
+    fn get_ids(&self, count: u64) -> Vec<Id> {
+        let start = self.next_id.fetch_add(count, Ordering::Relaxed);
+        let end = start + count;
+        (start..end).map(Id::Num).collect()
+    }
+
+    fn serialize_request(
+        &self,
+        input: MethodInvocation,
+    ) -> Result<SerializedRequest, RpcClientError> {
+        let id = Id::Num(self.next_id.fetch_add(1, Ordering::Relaxed));
+        Self::serialize_request_with_id(input, id)
+    }
+
+    fn serialize_request_with_id(
+        method: MethodInvocation,
+        id: Id,
+    ) -> Result<SerializedRequest, RpcClientError> {
+        let request = serde_json::to_value(&Request {
+            version: jsonrpc::Version::V2_0,
+            id,
+            method,
         })
-        .to_string();
+        .map_err(RpcClientError::InvalidJsonRequest)?;
 
-        SerializedRequest { id, request }
+        Ok(SerializedRequest(request))
     }
 
-    async fn call<T>(&self, input: &MethodInvocation) -> Result<T, RpcClientError>
-    where
-        T: for<'a> serde::Deserialize<'a>,
-    {
-        let cache_key = CacheableMethodInvocation::try_from(input)
+    async fn call<T: DeserializeOwned + Serialize>(
```

Comment:
Is there a reason that you're moving the bounds up here?

Reply:
We need `Serialize` here now because we extract the result from the response before writing it to the cache and as a consequence `write_response_to_cache` takes `impl Serialize` as argument instead of `&str`.  

(`DeserializeOwned` is just sugar for `for<'a> serde::Deserialize<'a>`)

Reply:
I was moreso referring to the move from the where clause to the generic type definition

Reply:
Ah just thought the where clause was an overkill for this. I typically use it with more complex generic bounds. Happy to put it back though